backbone.js

```diff
@@ -693,7 +718,12 @@
         urlError();
       if (this.isNew()) return base;
       var id = this.get(this.idAttribute);
-      return base.replace(/[^\/]$/, '$&/') + encodeURIComponent(id);
+      var consume = function(base) {
+        return base.replace(/[^\/]$/, '$&/') + encodeURIComponent(id);
+      };
+      if (base instanceof Promise)
+        return base.then(consume);
```

Comment:
Is 0d4fe5b23f1023df17b7e6b13c7f44639f64de41 what you had in mind?

Also, shall go ahead and drop the polyfill under `test/vendor`?